x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilder.java

```diff
@@ -222,7 +235,24 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
             );
         }
 
-        return (shouldPruneTokens)
+        // if the query options for pruning are not set,
+        // we need to check the index options for this field
+        // and use those if set - however, only if the index
+        // was created after we added this support.
+        if (context.indexVersionCreated().onOrAfter(SparseVectorFieldMapper.SPARSE_VECTOR_PRUNING_INDEX_OPTIONS_VERSION)) {
+            SparseVectorFieldMapper sparseVectorFieldMapper = getSparseVectorFieldMapperForSearchExecution(fieldName, context);
+            TokenPruningSet pruningOptions = setPruningConfigFromIndexIfNeeded(
+                shouldPruneTokens,
+                tokenPruningConfig,
+                sparseVectorFieldMapper
+            );
+
+            return pruningOptions.pruneTokens
```

Comment:
Can we consolidate this, so we're only doing this if check once? 

Check if it's specified/should default, and then return the appropriate query builder once after that?